modules/repository-gcs/build.gradle

```diff
@@ -26,9 +26,9 @@ esplugin {
 }
 
 dependencies {
-  api 'com.google.cloud:google-cloud-storage:1.118.1'
-  api 'com.google.cloud:google-cloud-core:2.0.2'
-  api 'com.google.cloud:google-cloud-core-http:2.0.2'
+  api 'com.google.cloud:google-cloud-storage:2.13.1'
```

Comment:
I believe you need to update the versions of all the transitive dependencies of `com.google.cloud:google-cloud-storage:2.13.1` which are

```
[INFO] com.google.cloud:google-cloud-storage:jar:2.13.0
[INFO] +- com.google.guava:guava:jar:31.1-jre:compile
[INFO] +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] +- org.checkerframework:checker-qual:jar:3.25.0:compile
[INFO] +- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] +- com.google.http-client:google-http-client:jar:1.42.2:compile
[INFO] +- io.opencensus:opencensus-contrib-http-util:jar:0.31.1:compile
[INFO] +- com.google.http-client:google-http-client-jackson2:jar:1.42.2:compile
[INFO] +- com.google.api-client:google-api-client:jar:2.0.0:compile
[INFO] +- com.google.oauth-client:google-oauth-client:jar:1.34.1:compile
[INFO] +- com.google.http-client:google-http-client-gson:jar:1.42.2:compile
[INFO] +- com.google.http-client:google-http-client-apache-v2:jar:1.42.2:compile
[INFO] +- com.google.apis:google-api-services-storage:jar:v1-rev20220705-2.0.0:compile
[INFO] +- com.google.code.gson:gson:jar:2.9.1:compile
[INFO] +- com.google.cloud:google-cloud-core:jar:2.8.20:compile
[INFO] +- com.google.auto.value:auto-value-annotations:jar:1.9:compile
[INFO] +- com.google.api.grpc:proto-google-common-protos:jar:2.9.6:compile
[INFO] +- com.google.cloud:google-cloud-core-http:jar:2.8.20:compile
[INFO] +- com.google.http-client:google-http-client-appengine:jar:1.42.2:compile
[INFO] +- com.google.api:gax-httpjson:jar:0.104.2:compile
[INFO] +- com.google.api:gax:jar:2.19.2:compile
[INFO] +- com.google.auth:google-auth-library-credentials:jar:1.11.0:compile
[INFO] +- com.google.auth:google-auth-library-oauth2-http:jar:1.11.0:compile
[INFO] +- com.google.api:api-common:jar:2.2.1:compile
[INFO] +- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO] +- io.opencensus:opencensus-api:jar:0.31.1:compile
[INFO] +- io.grpc:grpc-context:jar:1.49.2:compile
[INFO] +- com.google.api.grpc:proto-google-iam-v1:jar:1.6.2:compile
[INFO] +- com.google.protobuf:protobuf-java:jar:3.21.7:compile
[INFO] +- com.google.protobuf:protobuf-java-util:jar:3.21.7:compile
[INFO] +- org.threeten:threetenbp:jar:1.6.2:compile
[INFO] +- com.fasterxml.jackson.core:jackson-core:jar:2.13.4:compile
[INFO] \- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] -----------------------------------------------------
```

Reply:
```diff
Index: modules/repository-gcs/build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/modules/repository-gcs/build.gradle b/modules/repository-gcs/build.gradle
--- a/modules/repository-gcs/build.gradle	(revision 77e4646ec3d455ed7023ad3a36432b97507b9a27)
+++ b/modules/repository-gcs/build.gradle	(date 1671563198310)
@@ -27,35 +27,35 @@
 
 dependencies {
   api 'com.google.cloud:google-cloud-storage:2.13.1'
-  api 'com.google.cloud:google-cloud-core:2.8.28'
-  api 'com.google.cloud:google-cloud-core-http:2.8.28'
-  runtimeOnly 'com.google.guava:guava:30.1.1-jre'
+  api 'com.google.cloud:google-cloud-core:2.8.20'
+  api 'com.google.cloud:google-cloud-core-http:2.8.20'
+  runtimeOnly 'com.google.guava:guava:31.1-jre'
   runtimeOnly 'com.google.guava:failureaccess:1.0.1'
   api "commons-logging:commons-logging:${versions.commonslogging}"
   api "org.apache.logging.log4j:log4j-1.2-api:${versions.log4j}"
   api "commons-codec:commons-codec:${versions.commonscodec}"
   api 'com.google.api:api-common:2.2.1'
-  api 'com.google.api:gax:2.0.0'
-  api 'org.threeten:threetenbp:1.5.1'
+  api 'com.google.api:gax:2.19.2'
+  api 'org.threeten:threetenbp:1.6.2'
   api "com.google.protobuf:protobuf-java-util:${versions.protobuf}"
   api "com.google.protobuf:protobuf-java:${versions.protobuf}"
-  api 'com.google.code.gson:gson:2.8.9'
-  api 'com.google.api.grpc:proto-google-common-protos:2.3.2'
-  api 'com.google.api.grpc:proto-google-iam-v1:1.0.14'
-  api 'com.google.auth:google-auth-library-credentials:1.0.0'
-  api 'com.google.auth:google-auth-library-oauth2-http:1.0.0'
+  api 'com.google.code.gson:gson:2.9.1'
+  api 'com.google.api.grpc:proto-google-common-protos:2.9.6'
+  api 'com.google.api.grpc:proto-google-iam-v1:1.6.2'
+  api 'com.google.auth:google-auth-library-credentials:1.11.0'
+  api 'com.google.auth:google-auth-library-oauth2-http:1.11.0'
   api 'com.google.oauth-client:google-oauth-client:1.34.1'
-  api 'com.google.api-client:google-api-client:1.35.1'
-  api 'com.google.http-client:google-http-client:1.39.2'
-  api 'com.google.http-client:google-http-client-gson:1.39.2'
-  api 'com.google.http-client:google-http-client-appengine:1.39.2'
-  api 'com.google.http-client:google-http-client-jackson2:1.39.2'
+  api 'com.google.api-client:google-api-client:2.0.0'
+  api 'com.google.http-client:google-http-client:1.42.2'
+  api 'com.google.http-client:google-http-client-gson:1.42.2'
+  api 'com.google.http-client:google-http-client-appengine:1.42.2'
+  api 'com.google.http-client:google-http-client-jackson2:1.42.2'
   api "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
-  api 'com.google.api:gax-httpjson:0.85.0'
-  api 'io.grpc:grpc-context:1.39.0'
-  api 'io.opencensus:opencensus-api:0.28.0'
-  api 'io.opencensus:opencensus-contrib-http-util:0.28.0'
-  api 'com.google.apis:google-api-services-storage:v1-rev20210127-1.32.1'
+  api 'com.google.api:gax-httpjson:0.104.2'
+  api 'io.grpc:grpc-context:1.49.2'
+  api 'io.opencensus:opencensus-api:0.31.1'
+  api 'io.opencensus:opencensus-contrib-http-util:0.31.1'
+  api 'com.google.apis:google-api-services-storage:v1-rev20220705-2.0.0'
 
   testImplementation project(':test:fixtures:gcs-fixture')
 }

```

Reply:
I think some of your suggestions are actually downgrades! :-)
I've upgraded as many dependencies as I could in https://github.com/elastic/elasticsearch/pull/92327/commits/3e30165a816955711a032d34492b1c73cf96af87.

Reply:
I think we use to have a check that made sure that we pull the correct transitive dependencies, maybe it's not there anymore.